src/solvers/bicgstab_solver.hpp

```diff
@@ -251,9 +275,9 @@ class BiCGSTABSolver {
           solver->residual.val = 0.0;
           return TaskStatus::iterate;
         },
-        this, pmesh, params_.residual_tolerance);
+        this, pmesh, params_.max_iters, ptol);
 
-    return solver_id;
+    return tl.AddTask(solver_id, CopyData<x, u>, md);
```

Comment:
In the other included examples, I thought that an iterative TaskList should always finally return the `solver_id` (as opposed to the completion task or any other task).  Should `CopyData` be put inside the `CompletionTask` (if iterations complete), and then maintain `return solver_id;`?

Reply:
The way BiCGStab is now structured (which minimizes the number of fields that are communicated) requires adding a non-iterative task after the iterations (here copying the iteration output into the field defined by the downstream code). Notice that this task is in task list `tl` not `itl`.